File: HappyNumbers/happy.rb
#happy numbers soln in rb for codeeval.com by steven a dunn

def isHappy(n, visited)
    sum = 0
    n.split("").each do |c|
        sum += c.to_i * c.to_i
    end

    if sum == 1
        return true
    end

    if visited.include? sum
        return false;
    end

    visited << sum

    if isHappy(sum.to_s, visited)
        return true;
    end

    return false;
end


File.open(ARGV[0], 'r').each_line do |line|
    visited = []
    if isHappy(line.chomp, visited)
        puts 1
    else
        puts 0
    end
end

File: HappyNumbers/py2/happy.py
# Happy Numbers solution in Python 2 for CodeEval.com by Steven A. Dunn

def isHappy(n, visited):
  sum = 0
  for c in n:
    sum += int(c) * int(c)

  if sum == 1: return True
  if sum in visited: return False;
  visited.append(sum)

  if isHappy(str(sum), visited): return True;
  return False;

import sys

for line in open(sys.argv[1], 'r'):
  visited = list()
  if isHappy(line.rstrip('\n'), visited): print 1
  else: print 0
